File: examples/python/module_enumeration.py
from libmem import *

for module in LM_EnumModules():
    print(f"[*] Module Base: {module.base}")
    print(f"[*] Module End:  {module.end}")
    print(f"[*] Module Size: {module.size}")
    print(f"[*] Module Name: {module.name}")
    print(f"[*] Module Path: {module.path}")
    print("====================")

File: examples/c/module_enumeration.c
#include <libmem/libmem.h>

lm_bool_t
EnumModulesCallback(lm_module_t *pmod,
		    lm_void_t   *arg)
{
	printf("[*] Module Base: %p\n", (void *)pmod->base);
	printf("[*] Module End:  %p\n", (void *)pmod->end);
	printf("[*] Module Size: %p\n", (void *)pmod->size);
	printf("[*] Module Name: %s\n", pmod->name);
	printf("[*] Module Path: %s\n", pmod->path);
	printf("====================\n");

	return LM_TRUE;
}

int
main()
{
	LM_EnumModules(EnumModulesCallback, LM_NULL);

	return 0;
}
